style_editor.html

```diff
@@ -9912,15 +9967,111 @@
+
+          <button
+           title="Close Settings Panel."
+            class="close-settings" onclick="toggleSettingsPanel()">X</button>
+
+            <h2 class="settings-header-label">Settings</h2>
+
+          <div class="settings-section">
+            <h4 class="settings-section-label" title="These settings apply to the blade preview above.">Blade Preview:</h4>
+
+            <label title="Simulates In-Hilt LED blades, (no addressable pixel effects).">
+              <input type="checkbox" id="INHILT_BUTTON" name="inhilt" onclick="ClickInhilt()">
+              <span>Inhilt LED</span>
+            </label>
+
+            <label title="Provides &quot;snapshots&quot; of rapidly moving things to see what they look like without fast movement. *Note* Graflex hilt model probably won't be able to render with this on.">
+              <input type="checkbox" id="SLOW_BUTTON" name="slow" onclick="ClickSlow()">
+              <span>Slow</span>
+            </label>
+          </div>
+        </div>
+
+<script>
+
+function toggleSettingsPanel() {
+  var settingsButton = FIND("SETTINGS_BUTTON");
+  var settingsPanel = FIND("settings_panel");
+
+  settingsPanel.classList.toggle("show");
+
+  // Mouseleave event listener
+  if (settingsPanel.classList.contains("show")) {
+    var timeoutId = null;
+    settingsPanel.addEventListener("mouseleave", function(e) {
+      timeoutId = setTimeout(function() {
+        settingsPanel.classList.remove("show");
+      }, 1000);
+    });
+    settingsPanel.addEventListener("mouseenter", function(e) {
+      if (timeoutId) {
+        clearTimeout(timeoutId);
+        timeoutId = null;
+      }
+    });
+  }
+}
+
+/* Globally declare all the elements for Settings */
+var inhiltButton = FIND("INHILT_BUTTON"), inhiltState;
+var slowButton = FIND("SLOW_BUTTON"), slowState;
+
+/* Settings buttons saved as local storage */
+function getSavedState(buttonState, defaultValue) {
+  var value = localStorage.getItem(buttonState);
+  console.log("Retrieved SavedState for " + buttonState + ": " + value);
+
+  if (value === null) {
+    return defaultValue;
+  }
+  if (buttonState === "wavlenValue") {
+    return parseInt(value);
+  }
+  // Convert the string to boolean unless it's null or "false"
+  return value !== "false";
+}
+
+function saveState(buttonState, settingIsOn) {
+  localStorage.setItem(buttonState, settingIsOn);
+}
+
+// Call the onPageLoad function when the page is loaded
+window.addEventListener('DOMContentLoaded', onPageLoad);
+
+function onPageLoad() {
+  // INHILT_BUTTON
+  inhiltState = getSavedState("inhiltStateSave", false);
+  if (inhiltState) STATE_NUM_LEDS = 145 - STATE_NUM_LEDS;
+  inhiltButton.checked = inhiltState ? true : false;
+
+  // SLOW_BUTTON
+  slowState = getSavedState("slowStateSave", false);
+  if (slowState) framesPerUpdate = 60 - framesPerUpdate;
+  slowButton.checked = slowState ? true : false;
+}
+
+function ClickInhilt() {
+  inhiltState = !inhiltState;
```

Comment:
We shouldn't do three individual toggles, that makes it possible for these variables to get out of sync.
Instead of:

A=!A
B=!B
C=!C

we should do:

if (A) {
   A= false
   B=false
   C=false
} else {
  A = true
  B = true
  C = true
}

or possibly something like

A=!A
B=A
C=A

Reply:
sorry, which 3 toggles? there's one per button onclick.
And there are more buttons coming

Reply:
I wish thinking inhiltState, STATE_NUM_LEDS and inhiltButton.checked, but the last of them isn't toggled independently...